src/lib.rs

```diff
@@ -1815,6 +1815,16 @@ impl Connection {
             return Err(Error::Done);
         }
 
+        // Check if the peer has sent the stateless reset token. If yes, then
+        // move the connection to the draining state immediately.
+        // https://www.rfc-editor.org/rfc/rfc9000.html#section-10.3.1-5
+        if let Some(ref token) = self.local_transport_params.stateless_reset_token {
```

Comment:
From https://www.rfc-editor.org/rfc/rfc9000.html#section-10.3.1-2:
>  However, the comparison MUST be performed when the first packet in an incoming datagram either cannot be associated with a connection or cannot be decrypted.

so this should happen way later, and only after we failed to decrypt the packet, and also probably if e.g. decoding the header fails.

It might be easier to just move this to the `Err` case in `recv()`, and first check if `done == 0` (which means this is the first packet in the datagram).

Reply:
Thanks for the review. I'll fix that.

Reply:
By the way, it also has the following - https://www.rfc-editor.org/rfc/rfc9000.html#section-10.3.1-5
> If the last 16 bytes of the datagram are identical in value to a stateless reset token, the endpoint MUST enter the draining period and not send any further packets on this connection.

I interpreted it as - if the last 16 bytes are equal to the stateless reset token **no matter whether the packet could be processed or not**, then draining period should be entered.